extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufReader.java

```diff
@@ -217,7 +279,7 @@ private void addPointRows(Point point, Metric metric, LabelContext labelContext)
   }
 
   @Override
-  public CloseableIterator<InputRowListPlusRawValues> sample()
+  public CloseableIterator<InputRowListPlusRawValues> sample() throws IOException
```

Comment:
this change is unnecessary if we read the ByteBuffer directly as suggested above

Reply:
looks like we can avoid throwing checked exceptions here now

Reply:
This is thrown by the `read` method in the base interface. Doesn't have to do anything with the use of `open` after all. If we want to use try-with-resources which is good IMO, we need to add this exception to the method declaration. 

This is declared in: 
https://github.com/confluentinc/druid/blob/94001257689b8e423c438e6a9e2f4197529fc6aa/extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/data/input/kafkainput/KafkaInputReader.java#L143

while elsewhere the overriding method omits the exception in the signature. 

The interface is: https://github.com/confluentinc/druid/blob/0.22.1-confluent/core/src/main/java/org/apache/druid/data/input/InputEntityReader.java
and hasn't changed in recent versions.